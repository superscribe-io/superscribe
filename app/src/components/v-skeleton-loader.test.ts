import { test, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import VSkeletonLoader from './v-skeleton-loader.vue';

test('Mount component', () => {
	expect(VSkeletonLoader).toBeTruthy();

	const wrapper = mount(VSkeletonLoader, {
		props: {
			type: 'list-item-icon',
		},
	});

	expect(wrapper.html()).toMatchSnapshot();
});

test('type prop', () => {
	const types = ['input', 'input-tall', 'block-list-item', 'block-list-item-dense', 'text', 'list-item-icon'];

	for (const type of types) {
		const wrapper = mount(VSkeletonLoader, {
			props: {
				type,
			},
		});

		expect(wrapper.classes()).toContain(type);
	}
});
