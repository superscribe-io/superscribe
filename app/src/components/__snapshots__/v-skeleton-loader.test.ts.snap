// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Mount component 1`] = `
"<div data-v-b5d05879=\\"\\" class=\\"list-item-icon v-skeleton-loader\\">
  <div data-v-b5d05879=\\"\\" class=\\"icon\\"></div>
  <div data-v-b5d05879=\\"\\" class=\\"text\\"></div>
</div>"
`;
